### PR TITLE
[improvement](delete) support delete predicate on value column for merge-on-write unique table

### PR DESCRIPTION
**Pull request by bobhan1**
## Proposed changes
Previously, delete statement with conditions on value columns are only supported on duplicate tables. After we introduce delete sign mechanism to do batch delete, a delete statement with conditions on value columns on unique tables will be transformed into the corresponding `insert into ..., __DELETE_SIGN__ select ...` statement. However, for unique table with merge-on-write enabled, the overhead of inserting these data can be eliminated. So this PR add the ability to allow delete predicate on value columns for merge-on-write unique tables.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.75 seconds
 stream load tsv:          514 seconds loaded 74807831229 Bytes, about 138 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          66 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17161976403 Bytes

**Comment by bobhan1** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.9 seconds
 stream load tsv:          508 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17162175521 Bytes

**Comment by bobhan1** [comment]
run buildall

File: regression-test/suites/delete_p0/test_delete_on_value.groovy [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    sql """ CREATE TABLE ${tableName} (
+            `x` BIGINT NOT NULL,
+            `y` BIGINT NULL,
+            `z` BIGINT NULL)
+            ENGINE=OLAP
+            UNIQUE KEY(`x`)
+            COMMENT 'OLAP'
+            DISTRIBUTED BY HASH(`x`) BUCKETS 4
+            PROPERTIES (
+                "replication_num" = "1",
```
Comment (zhannngchen) [inline regression-test/suites/delete_p0/test_delete_on_value.groovy:34]: add case with seq column

**Comment by bobhan1** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.21 seconds
 stream load tsv:          512 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17162138876 Bytes

**Comment by bobhan1** [comment]
run p0

**Review by zhannngchen — APPROVED** [review]
LGTM

**Comment by bobhan1** [comment]
run p0

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**